nvflare/private/fed/server/server_engine.py

```diff
@@ -206,6 +206,16 @@ def remove_exception_process(self, job_id):
 
     def wait_for_complete(self, job_id, process):
         process.wait()
+        run_process_info = self.run_processes.get(job_id)
+        process_finished = run_process_info.get(RunProcessKey.PROCESS_FINISHED, False)
```

Comment:
this line `self.run_processes.get(job_id)` can return None.
then the following line will error.

Reply:
we only have one place to pop the "job_id" which is after the code here. We won't have the case of None.

Reply:
We have other places that will pop this job_id, for example: https://github.com/NVIDIA/NVFlare/blob/dev/nvflare/private/fed/server/server_engine.py#L319-L324

Reply:
Added the run_process_info None check.